spirit/comment/forms.py

```diff
@@ -150,13 +151,26 @@ def __init__(self, user=None, *args, **kwargs):
 
     def clean_file(self):
         file = self.cleaned_data['file']
+        file_mime = magic.from_buffer(file.read(), mime=True)
 
-        # Probably not reliable, but better than no validations
-        # https://stackoverflow.com/questions/6460848/how-to-limit-file-types-on-file-uploads-for-modelforms-with-filefields
-        if file.content_type not in settings.ST_ALLOWED_UPLOAD_FILE_FORMAT:
+        valid_ext = False
+        for file_ext in settings.ST_ALLOWED_UPLOAD_FILE_MEDIA_TYPE.keys():
+            if file.name.endswith(file_ext):
+                valid_ext = True
+
+        if valid_ext is False:
+            ext = file.name.split('.')[-1]
             raise forms.ValidationError(
-                _("Unsupported file format. Supported formats are %s."
-                  % ", ".join(settings.ST_ALLOWED_UPLOAD_FILE_FORMAT))
+                _("Unsupported file extension %s. Supported extensions are %s."
+                  % (ext, ", ".join(settings.ST_ALLOWED_UPLOAD_FILE_MEDIA_TYPE.keys()))
+                )
+            )
+
+        if file_mime not in settings.ST_ALLOWED_UPLOAD_FILE_MEDIA_TYPE.values():
+            raise forms.ValidationError(
+                _("Unsupported file mime type %s. Supported types are %s."
+                  % (file_mime, ", ".join(settings.ST_ALLOWED_UPLOAD_FILE_MEDIA_TYPE.values()))
+                )
```

Comment:
I *think* this can be simplified a bit (but maybe not XD) to:

```python
ext = os.path.splitext(file.name)[1].lstrip('.')  # Won't ever raise. Has at most one '.' so lstrip is fine here
mime = settings.ST_ALLOWED_UPLOAD_FILE_MEDIA_TYPE.get(ext, None)
if mime is None:
   raise forms.ValidationError('invalid extension')
if mime != file_mime:
   raise forms.ValidationError('invalid mime')
```

Reply:
Hmm.. you are right, the for loop is unnecessary (I am not sure why I added that there at the time). This is a better solution, thanks.